src/components/Chat/Chatroom.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import { throttle } from "lodash";
+
+import Loading from "../shared/Loading";
+import MessageBox from "./MessageBox";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+const ChatRoomContainer = styled.div`
+  width: 100%;
+  height: 90%;
+  overflow: auto;
+`;
+
+const TargetDiv = styled.div`
+  width: 100%;
+  height: 10px;
+`;
+
+const getMessages = async (goalId, nextPageToken) => {
+  const url = nextPageToken
+    ? `/api/chats/${goalId}?nextPageToken=${nextPageToken}`
+    : `/api/chats/${goalId}`;
+  const { data } = await axios.get(url);
+  return data.result;
+};
+
+export default function Chatroom() {
+  const scrollTarget = useRef();
+  const [target, setTarget] = useState(null);
+  const [messages, setMessages] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState("");
+  const { id } = useParams();
+
+  const getData = async (id, token) => {
+    const prevHeight = scrollTarget.current.scrollHeight;
+
+    try {
+      const { messages, nextPageToken } = await getMessages(id, token);
+      setMessages(prev => messages.concat(prev));
+      setToken(nextPageToken);
+      scrollTarget.current.scrollTop =
+        scrollTarget.current.scrollHeight - prevHeight;
+    } catch (error) {
+      console.log(error);
```

Comment:
```suggestion
      console.error(error);
```
console.log를 통해 error를 던져줄 수도 있지만, [console.error()](https://developer.mozilla.org/ko/docs/Web/API/Console/error)를 이용할 수도 있을 것 같습니다!